tests/test_execution.py

```diff
@@ -314,3 +314,75 @@ def test_aggregate_composite_traintuples(global_execution_env):
             traintuple=traintuple,
         )
         sessions[0].add_testtuple(spec).future().wait()
+
+
+@pytest.mark.parametrize('fail_count', [5, 6])
+def test_execution_retry_on_fail(fail_count, global_execution_env):
+    """Execution of a traintuple which fails on the N first tries, and suceeds on the N+1th try"""
+
+    # This test ensures the compute task retry mechanism works correctly.
+    #
+    # It executes an algorithm that `raise`s on the first N runs, and then
+    # succeeds.
+    #
+    # /!\ This test should ideally be part of the substra-backend project,
+    #     not substra-tests. For the sake of expendiency, we're keeping it
+    #     as part of substra-tests for now, but we intend to re-implement
+    #     it in substra-backend instead eventually.
+    # /!\ This test makes use of the "local" folder to keep track of a counter.
+    #     This is a hack to make the algo raise or succeed depending on the retry
+    #     count. Ideally, we would use a more elegant solution.
+    # /!\ This test doesn't validate that an error in the docker build phase (of
+    #     the compute task execution) triggers a retry. Ideally, it's that case that
+    #     would be tested, since errors in the docker build are the main use-case
+    #     the retry feature was build for.
+
+    RETRY_ALGO_SNIPPET_TOREPLACE = f"""
```

Comment:
formatting:
- no need to use capital letter here
- no need for a f-string

Reply:
Fixed: https://github.com/SubstraFoundation/substra-tests/pull/39/commits/9b245819deea49adc4ff0be51bf9bed4ce3dc38e